certbot/tests/cli_test.py

```diff
@@ -63,7 +65,8 @@ def _help_output(self, args):
         return output.getvalue()
 
     @mock.patch("certbot.cli.flag_default")
-    def test_cli_ini_domains(self, mock_flag_default):
+    @test_util.patch_get_utility()
```

Comment:
@bmw, I'm not sure if I understand step `3.`

`_help_output` calls `parse` (where `zope.component.getUtility` gets
mocked (step `2.`)); so, if I'm understanding it correctly, the
mocked `notification` is not available in `_help_output`.

How about this:

```diff
diff --git a/certbot/tests/cli_test.py b/certbot/tests/cli_test.py
index c4f9ad07..d19c15c2 100644
--- a/certbot/tests/cli_test.py
+++ b/certbot/tests/cli_test.py
@@ -2,6 +2,7 @@
 import argparse
 import unittest
 import os
+import sys
 import tempfile
 
 import mock
@@ -51,10 +52,23 @@ class ParseTest(unittest.TestCase):  # pylint: disable=too-many-public-methods
         reload_module(cli)
 
     @staticmethod
-    def parse(*args, **kwargs):
+    def _unmocked_parse(*args, **kwargs):
         """Get result of cli.prepare_and_parse_args."""
         return cli.prepare_and_parse_args(PLUGINS, *args, **kwargs)
 
+    @staticmethod
+    def parse(*args, **kwargs):
+        """Mocks zope.component.getUtility and calls _unmocked_parse."""
+        with test_util.patch_get_utility() as mocked_get_utility:
+            mocked_get_utility().notification.side_effect = \
+              ParseTest._stdout_notification
+            return ParseTest._unmocked_parse(*args, **kwargs)
+
+    @staticmethod
+    def _stdout_notification(*args, **kwargs):
+        """Send notification to stdout."""
+        sys.stdout.write(args[0] if args else kwargs['message'])
+
     def _help_output(self, args):
         "Run a command, and return the output string for scrutiny"
 
@@ -104,8 +118,8 @@ class ParseTest(unittest.TestCase):  # pylint: disable=too-many-public-methods
         self.assertEqual(namespace.chain_path, os.path.abspath(chain))
         self.assertEqual(namespace.fullchain_path, os.path.abspath(fullchain))
 
-    @test_util.patch_get_utility(stdout_notification=True)
-    def test_help(self, unused_mock_get_utility):
+
+    def test_help(self):
         self._help_output(['--help'])  # assert SystemExit is raised here
         out = self._help_output(['--help', 'all'])
```
`ParseTest.test_help` passes:

```zsh
(venv)  ~/[...]/certbot/certbot/tests
↳ python cli_test.py ParseTest.test_help
.
----------------------------------------------------------------------
Ran 1 test in 0.306s

OK